### ISSUE TITLE
[K-Bug] Expecting sort SortGeneratedTopCell but got SortK

### ISSUE THREAD

**Issue by Stevengre** [runtimeverification/k#2909]
### Which command

- [x] kompile
- [ ] kast
- [ ] krun
- [ ] kprove
- [ ] kprovex
- [ ] ksearch

### What K Version?

K version:    5.3.205

### Operating System

Linux

#### `error.k`
```k
module ERROR-SYNTAX
  imports STRING

  syntax Error ::= "#exception" "(" String ")"
endmodule

module ERROR
  imports ERROR-SYNTAX
  imports BOOL

  rule [exception-non-ex]: true ~> #exception(_S:String) => .K [anywhere]
  rule [exception-exists]: false ~> #exception(S:String) => #exception(S) [anywhere]
endmodule
```

#### `test.k`
```k
requires "error.k"

module TEST
  imports ERROR
endmodule
```

### Steps to Reproduce

```shell
kompile test.k --backend haskell
```

### Expected Results

**now:**
kore-exec: [85986] Error (ErrorVerify):
    Error:
      module 'HCSP2SEQUENCE':
      alias 'rule0LHS' declaration (definition.kore 398:9):
      \and (definition.kore 400:10):
      (definition.kore 400:10, definition.kore 10:43):
        Expecting sort SortGeneratedTopCell{} but got SortK{}.
Created bug report: kore-exec.tar.gz
[Error] Critical: Haskell backend reported errors validating compiled
definition.
Examine output to see errors.

**exepected:**

**more info:** right after delete anywhere

**Comment by Baltoli** [comment]
I don't think that you can write `[anywhere]` rules over K sequences. This code is kompiled fine by the LLVM backend, but doesn't actually apply the rules as written even when the configuration matches.

If the Haskell backend rejects this at validation time, and the LLVM backend accepts it but won't apply the rules, I think this ought to be a frontend check. @radumereuta what do you think?

@Stevengre If you give us some more context about what you're trying to write here, we can help you do so.

**Comment by radumereuta** [comment]
If we add a check to the frontend, it might get out of sync with what the backends can support.
What are the current abilities of the backends regarding `anywhere` rules?
@dwightguth or @theo25 
@ana-pantilie

**Comment by ana-pantilie** [comment]
Anywhere rules are currently not supported in the Haskell backend.

**Comment by Stevengre** [comment]
Thank you so much!

**Comment by radumereuta** [comment]
Add a check in the front-end for anywhere rules which are not supported in the haskell backend.

**Comment by radumereuta** [comment]
Make sure to test the C semantics this time: https://github.com/runtimeverification/k/pull/3041

**Comment by Baltoli** [comment]
Yes, apologies this got missed the first time @radumereuta!

**Comment by radumereuta** [comment]
> Yes, apologies this got missed the first time @radumereuta!

What? I made the PR, I was supposed to test it.

**Comment by ehildenb** [comment]
We'll re-implement this change in the following way:

- Frontend should not allow `anywhere` rules going to Haskell backend.
- Frontend should have a flag `--allow-anywhere-haskell` which tells it to turn the prior check off, and will instead:
  - Print out a loud warning that `anywhere` rules are not allowed for haskell backend.
  - Strip out that rule before sending the definition to the backend.

**Comment by Baltoli** [comment]
Closed by #3051
